azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/validator/V2ConfigurationValidator.java

```diff
@@ -0,0 +1,97 @@
+
+import com.microsoft.azure.management.appservice.JavaVersion;
+import com.microsoft.azure.management.appservice.RuntimeStack;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.azure.maven.webapp.AbstractWebAppMojo;
+import com.microsoft.azure.maven.webapp.configuration.RuntimeSetting;
+import org.codehaus.plexus.util.StringUtils;
+
+import java.util.Arrays;
+
+import static com.microsoft.azure.maven.webapp.configuration.RuntimeSetting.RUNTIME_CONFIG_REFERENCE;
+
+public class V2ConfigurationValidator extends ConfigurationValidator {
+
+    public static final String[] VALID_OS = new String[]{"windows", "linux", "docker"};
+
+    public V2ConfigurationValidator(AbstractWebAppMojo mojo) {
+        super(mojo);
+    }
+
+    @Override
+    public String validateRegion() {
+        final String region = mojo.getRegion();
+        if (!StringUtils.isEmpty(region) && Region.findByLabelOrName(region) == null) {
+            return "The value of <region> is not supported, please correct it in pom.xml.";
+        }
+        return null;
+    }
+
+    @Override
+    public String validateOs() {
+        final RuntimeSetting runtime = mojo.getRuntime();
+        final String os = runtime.getOs();
+        if (runtime.isEmpty()) {
+            return null;
+        } else if (StringUtils.isEmpty(os)) {
+            return "Pleas configure the <os> of <runtime> in pom.xml.";
+        }
+        if (!Arrays.asList(VALID_OS).contains(os)) {
+            return "The value of <os> is not correct, supported values are: windows, linux and docker.";
+        }
+        return null;
+    }
+
+    @Override
+    public String validateRuntimeStack() {
+        final RuntimeSetting runtime = mojo.getRuntime();
+        if (runtime == null || runtime.isEmpty()) {
+            return null;
+        }
+        final RuntimeStack result = runtime.getLinuxRuntime();
+        return result == null ? String.format("Unsupported values for linux runtime, please refer %s " +
+                "more information", RUNTIME_CONFIG_REFERENCE) : null;
+    }
+
+    @Override
+    public String validateJavaVersion() {
+        final RuntimeSetting runtime = mojo.getRuntime();
+        if (runtime == null) {
+            return "Pleas config the <runtime> in pom.xml.";
+        }
+        if (runtime.getJavaVersion() == null || !JavaVersion.values().contains(runtime.getJavaVersion())) {
+            return "The configuration <javaVersion> in pom.xml is not correct.";
+        }
+        return null;
+    }
+
+    @Override
+    public String validateWebContainer() {
+        final RuntimeSetting runtime = mojo.getRuntime();
+        if (runtime == null) {
+            return "Pleas config the <runtime> in pom.xml.";
+        }
+        if (runtime.getWebContainer() == null) {
+            return "The configuration <webContainer> in pom.xml is not correct.";
+        }
+        return null;
+    }
+
+    public String validateImage() {
```

Comment:
two validateImage with different logic?

Reply:
Yes, for V1 schema the parent node of `<image>` is `<containerSettings>` while V2 is `<runtime>`